File: CS-Examples/11_Formatting/UsePredefinedStyles.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core.Spreadsheet;

namespace UsePredefinedStyles
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            // Create a new style
            CellStyle style = workbook.Styles.Add("newStyle");
            style.Font.FontName = "Calibri";
            style.Font.IsBold = true;
            style.Font.Size = 15;
            style.Font.Color = Color.CornflowerBlue;

            // Get the "B5" cell
            CellRange range = sheet.Range["B5"];
            range.Text = "Welcome to use Spire.XLS";

            // Apply the newly created style to the cell
            range.CellStyleName = style.Name;

            // Autofit the columns for better display of cell content
            range.AutoFitColumns();

            // Specify the name for the resulting Excel file
            String result = "UsePredefinedStyles_result.xlsx";

            // Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}

File: CS-Examples/11_Formatting/UsePredefinedStyles.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core.Spreadsheet

Namespace UsePredefinedStyles
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Create a new cell style named "newStyle"
            Dim style As CellStyle = workbook.Styles.Add("newStyle")
            style.Font.FontName = "Calibri"
            style.Font.IsBold = True
            style.Font.Size = 15
            style.Font.Color = Color.CornflowerBlue

            ' Get the cell range B5
            Dim range As CellRange = sheet.Range("B5")

            ' Set the text of the cell and apply the "newStyle" to it
            range.Text = "Welcome to use Spire.XLS"
            range.CellStyleName = style.Name

            ' Auto-fit the columns in the range
            range.AutoFitColumns()

            ' Save the modified workbook to a new file
            Dim result As String = "UsePredefinedStyles_result.xlsx"
            workbook.SaveToFile(result, ExcelVersion.Version2010)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
